### ISSUE TITLE
Implement interactive command-variable resolver

### ISSUE THREAD

**Issue by a1994846931931** [theia-ide/theia#5983]
### Description
In VSCode, it is possible to interactively ask for the program path by
1. Set `program` to `"${workspaceFolder}/${command:askProgramPath}"` in `launch.json`
2. Register `askProgramPath` command in plugin, for example
```typescript
'cdt.debug.askProgramPath', () => {
	return vscode.window.showInputBox({
		placeHolder: "Please enter the path to the program",
	})
}
```

When debugging, an input box will be shown, and `${command:askProgramPath}` will be replaced by the user input. 

Do we have a plan for implementing this feature in theia?

### Reproduction Steps
1. Build theia with [theia-cpp-extension](https://github.com/eclipse/theia-cpp-extension)
2. Add the following configuration to `launch.json`
```json
{
    "type": "gdb",
    "name": "Default",
    "request": "launch",
    "program": "${workspaceFolder}/${command:askProgramPath}",
}
```
3. Run debug

**OS and Theia version:**

**Diagnostics:**

**Comment by akosyakov** [comment]
@a1994846931931 I believe it was already implemented for launch configurations. Could you try with the latest master? See commits https://github.com/theia-ide/theia/pull/5835/commits/910dd8ead2d0dde900951e4503c0382e558d56b7 and https://github.com/theia-ide/theia/pull/5835/commits/2c97bc588c19de816350c51734675f5fdcfbe05d

**Comment by akosyakov** [comment]
although variable looks bogus, should not it be `command:cdt.debug.askProgramPath`?

**Comment by a1994846931931** [comment]
> btw variable looks bogus, should not it be `command:cdt.debug.askProgramPath`?

@akosyakov Yes, it works if I change `command:askProgramPath` to `command:cdt.debug.askProgramPath`. But it seems that VSCode is able to map `command:askProgramPath` to `command:cdt.debug.askProgramPath` automatically. So I guess the real problem is that theia cannot do the command mapping? Not very sure about this. Please tell me if I am wrong.

**Comment by akosyakov** [comment]
@a1994846931931 I see i did not it, if you can dig a bit in VS Code docs to find the contract for such shortcuts, it would be helpful. We need to fix then indeed.
